### ISSUE TITLE
Logrotate Error

### ISSUE THREAD

**Issue by mindrunner** [tomav/docker-mailserver#1507]
Got this today:

```
/etc/cron.daily/logrotate:
logrotate_script: 6: logrotate_script: Syntax error: "fi" unexpected
error: error running non-shared postrotate script for /var/log/mail/freshclam.log of '/var/log/mail/freshclam.log '
run-parts: /etc/cron.daily/logrotate exited with return code 1
```

**Comment by youtous** [comment]
What is the content of your `/etc/cron.daily/logrotate`?

Is there any error when running manually `/usr/sbin/logrotate /etc/logrotate.conf`?

**Comment by mindrunner** [comment]
```
#!/bin/sh

# skip in favour of systemd timer
if [ -d /run/systemd/system ]; then
    exit 0
fi

# this cronjob persists removals (but not purges)
if [ ! -x /usr/sbin/logrotate ]; then
    exit 0
fi

/usr/sbin/logrotate /etc/logrotate.conf
EXITVALUE=$?
if [ $EXITVALUE != 0 ]; then
    /usr/bin/logger -t logrotate "ALERT exited abnormally with [$EXITVALUE]"
fi
exit $EXITVALUE
```
```
root@mx0:/# /usr/sbin/logrotate /etc/logrotate.conf
root@mx0:/# 
```

Seems all good here. Normally it is working fine. Just last night it seemed to fail.

**Comment by erik-wramner** [comment]
I think my fix was less than stellar, but the error only occurs when the freshclam logs have been rotated. I commented out the postrotate action, but only a comment inside an if statement is apparently not correct. This can be ignored (the file has already been rotated) but it should of course be fixed.

**Comment by mindrunner** [comment]
happened again today

![image](https://user-images.githubusercontent.com/1413542/82143372-ab285780-9843-11ea-80a1-4173a7695b8f.png)

**Comment by gmasse** [comment]
I think the issue is in `/etc/logrotate.d/clamav-freshclam`
```
/var/log/mail/freshclam.log {
     rotate 12
     weekly
     compress
     delaycompress
     missingok
     create 640  clamav adm
     postrotate
     if [ -d /run/systemd/system ]; then
         systemctl -q is-active clamav-freshclam && systemctl kill --signal=SIGHUP clamav-freshclam || true
     else
         #invoke-rc.d clamav-freshclam reload-log > /dev/null || true
     fi
     endscript
     }
```

If I compare with `clamav-daemon`, if suppose the commented line would be replaced by:
```
         /usr/bin/supervisorctl signal hup clamav-freshclam >/dev/null || true
```

**Comment by mindrunner** [comment]
Alright. Is the directory `/run/systemd/system` supposed to be there? 

We should eliminate the whole `else` block. It doesn't do anything, right?

**Comment by erik-wramner** [comment]
Freshclam runs from cron, not as a daemon. The whole postrotate block (from postrotate to endscript?) should be removed, it is not needed.

**Comment by mindrunner** [comment]
@gmasse You were faster than me! Thanks for fixing! :)

**Comment by helmutundarnold** [comment]
Hello,
i have the same problem. Since 3 weeks I get a cron error mail with subject

           "Cron <root@mail> test -x /usr/sbin/anacron || ( cd / &&
run-parts --report /etc/cron.daily )"

and content says

/etc/cron.daily/logrotate:
logrotate_script: 6: logrotate_script: Syntax error: "fi" unexpected
error: error running non-shared postrotate script for /var/log/mail/freshclam.log of '/var/log/mail/freshclam.log '
run-parts: /etc/cron.daily/logrotate exited with return code 1

Problem is this commented line. Today I've run the update with lastet
but its still commented.


Regards
      Helmut


Am 17.05.20 um 18:18 schrieb Germain Masse:

**Comment by erik-wramner** [comment]
@helmutundarnold this has been fixed in latest and in 7.0.1. Perhaps you didn't recreate the container (compose down/up) or something? This is the logrotate job I have:

```
/var/log/mail/freshclam.log {
     rotate 12
     weekly
     compress
     delaycompress
     missingok
     create 640  clamav adm
     }
```

As you can see there is no commented invoke-rc.d left.

**Comment by helmutundarnold** [comment]
Thanks for this quick Answer Erik,

indeed I've just run

 docker-compose -f <path to>/docker-compose.yml -p <container>  pull

After recreating the container  its solved.

Thanks for all your Support

Best Regards
      Helmut

Am 29.05.20 um 07:36 schrieb Erik Wramner:
